packages/stores/src/account/index.ts

```diff
@@ -1565,6 +1566,446 @@ export class OsmosisAccountImpl {
     );
   }
 
+  /**
+   * Migrates **unlocked** shares to full range concentrated position.
+   *
+   * @param poolId Id of pool to exit.
+   * @param gammShares Amount of gamm shares to migrate.
+   * @param memo Transaction memo.
+   * @param onFulfill Callback to handle tx fullfillment given raw response.
+   */
+  async sendMigrateUnlockedSharesToFullRangeConcentratedPositionMultiMsg(
```

Comment:
These functions seem exceptionally long/complex to me. Is there a way to break them down into more manageable chunks?

Reply:
Also, it appears they are untested?

Reply:
Yeah a few things here:
* There's a lot of code because there's a lot to do, I've tried to make it more scannable by adding commented sections that annotate what's happening. But IMO stuff like this is bespoke to the specific multi msg that's being crafted and is not expected to be reused at all. I think there's a lot of value in less components floating around.

* It's not tested because there needs to be a link between the balancer <> CL pool set up via a gov prop. I tried to craft that message just for the e2e test but I gave up because I started to get very cryptic errors about it not being created properly, I moved onto higher priority stuff

* I unfortunately just found out, but I was told the `sendUnlockAndMigrateSharesToFullRangeConcentratedPositionMsg` message below also handles this case that I implemented myself manually: if you pass a lockId of 0 and the gamm shares it will actually exit the pool for you before creating the CL position. In conclusion, this will all be deleted since this case is handled on chain.

Reply:
Will remove this message